src/mame/namco/namcos10_exio.cpp

```diff
@@ -0,0 +1,210 @@
+// license:BSD-3-Clause
+// copyright-holders:windyfairy
+#include "emu.h"
+#include "namcos10_exio.h"
+
+#include "logmacro.h"
+
+DEFINE_DEVICE_TYPE(NAMCOS10_EXIO,      namcos10_exio_device,      "namcos10_exio",   "Namco System 10 EXIO")
+DEFINE_DEVICE_TYPE(NAMCOS10_EXIO_BASE, namcos10_exio_base_device, "namcos10_exio_g", "Namco System 10 EXIO(G)")
+DEFINE_DEVICE_TYPE(NAMCOS10_MGEXIO,    namcos10_mgexio_device,    "namcos10_mgexio", "Namco System 10 MGEXIO")
+
+// EXIO(G) has the bare minimum: CPLD, audio output jacks, gun I/O, and a card edge connector for additional I/O
+namcos10_exio_base_device::namcos10_exio_base_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock, uint8_t ident_code)
+	: device_t(mconfig, type, tag, owner, clock), m_ident_code(ident_code)
+{
+}
+
+namcos10_exio_base_device::namcos10_exio_base_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: namcos10_exio_base_device(mconfig, NAMCOS10_EXIO_BASE, tag, owner, clock, 0x32)
+{
+}
+
+////////////////////////////////////
+
+namcos10_exio_device::namcos10_exio_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: namcos10_exio_base_device(mconfig, NAMCOS10_EXIO, tag, owner, clock, 0x30),
+	m_maincpu(*this, "exio_mcu"),
+	m_ram(*this, "exio_ram")
+{
+}
```

Comment:
If you’re putting the commas at the ends of the lines in initialiser lists, please put the colons introducing the initialiser lists at the end of the line as well, like:
```c++
namcos10_exio_device::namcos10_exio_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
	namcos10_exio_base_device(mconfig, NAMCOS10_EXIO, tag, owner, clock, 0x30),
	m_maincpu(*this, "exio_mcu"),
	m_ram(*this, "exio_ram")
{
}
```